### PR TITLE
Restore the original env-var value before asserting.

### PR DESCRIPTION
**Pull request by tonyredondo**
Restore the original env-var value before asserting. To avoid any collision to CI App dogfooding

@DataDog/apm-dotnet

### CODE REVIEW

**Comment by andrewlock** [comment]
## Code Coverage Report :bar_chart:

:heavy_check_mark: Merging #1816 into [master](https://github.com/DataDog/dd-trace-dotnet/tree/fbcfa7d1bbd71142c656321fda1cad63f55a0220) will **not change** line coverage
:heavy_check_mark: Merging #1816 into [master](https://github.com/DataDog/dd-trace-dotnet/tree/fbcfa7d1bbd71142c656321fda1cad63f55a0220) will **not change** branch coverage
:heavy_check_mark: Merging #1816 into [master](https://github.com/DataDog/dd-trace-dotnet/tree/fbcfa7d1bbd71142c656321fda1cad63f55a0220) will **not change** complexity


|           | [master](https://github.com/DataDog/dd-trace-dotnet/tree/fbcfa7d1bbd71142c656321fda1cad63f55a0220) | #1816       | Change   | 
|:----------|:-----------:|:-----------:|:--------:|
| Lines     | `9429` / `15031` | `9429` / `15031` |          |
| Lines %   | `63%`      | `63%`      |  `0%` :heavy_check_mark:  |
| Branches  | `4160` / `6510` | `4159` / `6510` |          |
| Branches %| `64%`      | `64%`      |  `0%` :heavy_check_mark:  |
| Complexity|   `7672`      | `7672`        |  `0`  :heavy_check_mark:    |

View the full report for further details:

* [HTML report for master](https://dev.azure.com/datadoghq/dd-trace-dotnet/_build/results?buildId=75857&view=codecoverage-tab) | [Source file](https://dev.azure.com/datadoghq/a51c4863-3eb4-4c5d-878a-58b41a049e4e/_apis/build/builds/75857/artifacts?artifactName=Code%20Coverage%20Report_75857&api-version=6.0&%24format=zip)  
* [HTML report for this PR #1816](https://dev.azure.com/datadoghq/dd-trace-dotnet/_build/results?buildId=75879&view=codecoverage-tab) | [Source file](https://dev.azure.com/datadoghq/a51c4863-3eb4-4c5d-878a-58b41a049e4e/_apis/build/builds/75879/artifacts?artifactName=Code%20Coverage%20Report_75879&api-version=6.0&%24format=zip)

### Datadog.Trace Breakdown :heavy_check_mark:

|        | [master](https://github.com/DataDog/dd-trace-dotnet/tree/fbcfa7d1bbd71142c656321fda1cad63f55a0220) | #1816       | Change   | 
|:-------|:-----------:|:-----------:|:--------:|
| Lines %| `63%`      | `63%`       |  `0%` :heavy_check_mark:  |
| Branches %| `64%`      | `64%`       |  `0%` :heavy_check_mark:  |
| Complexity| `7672`      | `7672`       |  `0` :heavy_check_mark:  |

The following classes have significant coverage changes.

| File    | Line coverage change | Branch coverage change | Complexity change |
|:--------|:--------------------:|:----------:|:--------:|
| [Datadog.Trace.Logging.LogRateLimiter](https://github.com/DataDog/dd-trace-dotnet/tree/c380cf4c420b75d519048432366e75e27c6c6107/tracer/src/Datadog.Trace/Logging/LogRateLimiter.cs) | `0%` :heavy_check_mark: | `-17%` :no_entry:   | `0` :heavy_check_mark: |
| [Datadog.Trace.RuntimeMetrics.RuntimeMetricsWriter](https://github.com/DataDog/dd-trace-dotnet/tree/c380cf4c420b75d519048432366e75e27c6c6107/tracer/src/Datadog.Trace/RuntimeMetrics/RuntimeMetricsWriter.cs) | `0%` :heavy_check_mark: | `8%` :heavy_check_mark:   | `0` :heavy_check_mark: |


View the full reports for further details:

* [HTML report for master](https://dev.azure.com/datadoghq/dd-trace-dotnet/_build/results?buildId=75857&view=codecoverage-tab) | [Source file](https://dev.azure.com/datadoghq/a51c4863-3eb4-4c5d-878a-58b41a049e4e/_apis/build/builds/75857/artifacts?artifactName=Code%20Coverage%20Report_75857&api-version=6.0&%24format=zip)  
* [HTML report for this PR #1816](https://dev.azure.com/datadoghq/dd-trace-dotnet/_build/results?buildId=75879&view=codecoverage-tab) | [Source file](https://dev.azure.com/datadoghq/a51c4863-3eb4-4c5d-878a-58b41a049e4e/_apis/build/builds/75879/artifacts?artifactName=Code%20Coverage%20Report_75879&api-version=6.0&%24format=zip)

File: tracer/test/Datadog.Trace.Tests/Configuration/ConfigurationSourceTests.cs [hunk 1]
```diff
@@ -244,14 +247,17 @@ public void GlobalDefaultSetting(Func<GlobalSettings, object> settingGetter, obj
             Func<GlobalSettings, object> settingGetter,
             object expectedValue)
         {
+            IConfigurationSource source = new EnvironmentConfigurationSource();
+
             // save original value so we can restore later
             var originalValue = Environment.GetEnvironmentVariable(key);
+            Environment.SetEnvironmentVariable(key, string.Empty, EnvironmentVariableTarget.Process);
```
Comment (andrewlock) [inline tracer/test/Datadog.Trace.Tests/Configuration/ConfigurationSourceTests.cs:254]: This line seems unnecessary seeing as it's immediately changed on the next line?

File: tracer/test/Datadog.Trace.Tests/Configuration/ConfigurationSourceTests.cs [hunk 2]
```diff
@@ -180,19 +180,22 @@ public void DefaultSetting(Func<TracerSettings, object> settingGetter, object ex
...
+                Environment.SetEnvironmentVariable(key, originalValue, EnvironmentVariableTarget.Process);
             }
             else
             {
                 Environment.SetEnvironmentVariable(key, value, EnvironmentVariableTarget.Process);
                 IConfigurationSource source = new EnvironmentConfigurationSource();
                 settings = new TracerSettings(source);
+
+                // restore original value
+                Environment.SetEnvironmentVariable(key, originalValue, EnvironmentVariableTarget.Process);
```
Comment (andrewlock) [inline tracer/test/Datadog.Trace.Tests/Configuration/ConfigurationSourceTests.cs:194]: Both branches call this, can we move it below the `if..else..` to make that a bit more obvious?

**Review by andrewlock — APPROVED** [review]
Just a couple of nits

File: tracer/test/Datadog.Trace.Tests/Configuration/ConfigurationSourceTests.cs (hunk 1 above)
Comment (tonyredondo) [reply tracer/test/Datadog.Trace.Tests/Configuration/ConfigurationSourceTests.cs]: you're right, done in f55b55d

File: tracer/test/Datadog.Trace.Tests/Configuration/ConfigurationSourceTests.cs (hunk 2 above)
Comment (tonyredondo) [reply tracer/test/Datadog.Trace.Tests/Configuration/ConfigurationSourceTests.cs]: you're right, done in f55b55d

File: tracer/test/Datadog.Trace.Tests/Configuration/ConfigurationSourceTests.cs [hunk 3]
```diff
@@ -174,25 +174,39 @@ public void DefaultSetting(Func<TracerSettings, object> settingGetter, object ex
...
             }
+            else if (key == ConfigurationKeys.AgentHost || key == ConfigurationKeys.AgentPort)
+            {
+                // We need to ensure DD_TRACE_AGENT_URL is empty.
+                string originalAgentUri = Environment.GetEnvironmentVariable(ConfigurationKeys.AgentUri);
+                Environment.SetEnvironmentVariable(ConfigurationKeys.AgentUri, null, EnvironmentVariableTarget.Process);
+
+                settings = GetTracerSettings(key, value);
+
+                // after load settings we can restore the original DD_SERVICE
```
Comment (andrewlock) [inline tracer/test/Datadog.Trace.Tests/Configuration/ConfigurationSourceTests.cs:190]:
```suggestion
                // after load settings we can restore the original DD_TRACE_AGENT_URL
```

**Comment by andrewlock** [comment]
## Code Coverage Report :bar_chart:

:heavy_check_mark: Merging #1816 into [master](https://github.com/DataDog/dd-trace-dotnet/tree/ea29038f536d0600c68597e78f1bba747a504c10) will **not change** line coverage
:heavy_check_mark: Merging #1816 into [master](https://github.com/DataDog/dd-trace-dotnet/tree/ea29038f536d0600c68597e78f1bba747a504c10) will **not change** branch coverage
:heavy_check_mark: Merging #1816 into [master](https://github.com/DataDog/dd-trace-dotnet/tree/ea29038f536d0600c68597e78f1bba747a504c10) will will **decrease** complexity by `12`


|           | [master](https://github.com/DataDog/dd-trace-dotnet/tree/ea29038f536d0600c68597e78f1bba747a504c10) | #1816       | Change   | 
|:----------|:-----------:|:-----------:|:--------:|
| Lines     | `9431` / `15040` | `9428` / `15031` |          |
| Lines %   | `63%`      | `63%`      |  `0%` :heavy_check_mark:  |
| Branches  | `4161` / `6522` | `4158` / `6510` |          |
| Branches %| `64%`      | `64%`      |  `0%` :heavy_check_mark:  |
| Complexity|   `7684`      | `7672`        |  `-12`  :heavy_check_mark:    |

View the full report for further details:

* [HTML report for master](https://dev.azure.com/datadoghq/dd-trace-dotnet/_build/results?buildId=75953&view=codecoverage-tab) | [Source file](https://dev.azure.com/datadoghq/a51c4863-3eb4-4c5d-878a-58b41a049e4e/_apis/build/builds/75953/artifacts?artifactName=Code%20Coverage%20Report_75953&api-version=6.0&%24format=zip)  
* [HTML report for this PR #1816](https://dev.azure.com/datadoghq/dd-trace-dotnet/_build/results?buildId=75918&view=codecoverage-tab) | [Source file](https://dev.azure.com/datadoghq/a51c4863-3eb4-4c5d-878a-58b41a049e4e/_apis/build/builds/75918/artifacts?artifactName=Code%20Coverage%20Report_75918&api-version=6.0&%24format=zip)

### Datadog.Trace Breakdown :heavy_check_mark:

|        | [master](https://github.com/DataDog/dd-trace-dotnet/tree/ea29038f536d0600c68597e78f1bba747a504c10) | #1816       | Change   | 
|:-------|:-----------:|:-----------:|:--------:|
| Lines %| `63%`      | `63%`       |  `0%` :heavy_check_mark:  |
| Branches %| `64%`      | `64%`       |  `0%` :heavy_check_mark:  |
| Complexity| `7684`      | `7672`       |  `-12` :heavy_check_mark:  |

The following classes have significant coverage changes.

| File    | Line coverage change | Branch coverage change | Complexity change |
|:--------|:--------------------:|:----------:|:--------:|
| [Datadog.Trace.ClrProfiler.CallTarget.Handlers.Continuations.TaskContinuationGenerator`4](https://github.com/DataDog/dd-trace-dotnet/tree/3963861d84dcef4ed08e701fe3681b31946ea860/tracer/src/Datadog.Trace/ClrProfiler/CallTarget/Handlers/Continuations/TaskContinuationGenerator`1.cs) | `-5%` :no_entry: | `-9%` :no_entry:   | `0` :heavy_check_mark: |
| [Datadog.Trace.Logging.LogRateLimiter](https://github.com/DataDog/dd-trace-dotnet/tree/3963861d84dcef4ed08e701fe3681b31946ea860/tracer/src/Datadog.Trace/Logging/LogRateLimiter.cs) | `0%` :heavy_check_mark: | `-17%` :no_entry:   | `0` :heavy_check_mark: |
| [Datadog.Trace.RuntimeMetrics.RuntimeMetricsWriter](https://github.com/DataDog/dd-trace-dotnet/tree/3963861d84dcef4ed08e701fe3681b31946ea860/tracer/src/Datadog.Trace/RuntimeMetrics/RuntimeMetricsWriter.cs) | `0%` :heavy_check_mark: | `-8%` :no_entry:   | `0` :heavy_check_mark: |


View the full reports for further details:

* [HTML report for master](https://dev.azure.com/datadoghq/dd-trace-dotnet/_build/results?buildId=75953&view=codecoverage-tab) | [Source file](https://dev.azure.com/datadoghq/a51c4863-3eb4-4c5d-878a-58b41a049e4e/_apis/build/builds/75953/artifacts?artifactName=Code%20Coverage%20Report_75953&api-version=6.0&%24format=zip)  
* [HTML report for this PR #1816](https://dev.azure.com/datadoghq/dd-trace-dotnet/_build/results?buildId=75918&view=codecoverage-tab) | [Source file](https://dev.azure.com/datadoghq/a51c4863-3eb4-4c5d-878a-58b41a049e4e/_apis/build/builds/75918/artifacts?artifactName=Code%20Coverage%20Report_75918&api-version=6.0&%24format=zip)

**Review by zacharycmontoya — APPROVED** [review]

**Pull request merged**